OSDOCS#13277: Clarify certificate rotation, including not rotating CAs (#91079)
openshift/openshift-docs: modules/microshift-certificate-lifetime.adoc

. Short-lived certificates having certificate validity of one year.
. Long-lived certificates having certificate validity of 10 years.

Most server or leaf certificates are short-lived.

An example of a long-lived certificate is the client certificate for `system:admin user` authentication, or the certificate of the signer of the `kube-apiserver` external serving certificate.

[id="microshift-certificate-rotation_{context}"]
== Certificate rotation
Certificates that are expired or close to their expiration dates need to be rotated to ensure continued {microshift-short} operation. When {microshift-short} restarts for any reason, certificates that are close to expiring are rotated. A certificate that is set to expire imminently, or has expired, can cause an automatic {microshift-short} restart to perform a rotation.

[IMPORTANT]
====

> If the rotated certificate is a {microshift-short} certificate authority (CA) then all of the signed certificates rotate. If you created any custom CAs, ensure the CAs manually rotate.

Reviewer: You should put a comma before the “then” in if/then sentences.
```suggestion
If the rotated certificate is a {microshift-short} certificate authority (CA), then all of the signed certificates rotate. If you created any custom CAs, ensure the CAs manually rotate.
```

> If the rotated certificate is a {microshift-short} Certificate Authority, all of the certificates it signed rotate. If you created any Custom Certificate Authorities, they must be rotated manually.

Reviewer 2: ```suggestion
If the rotated certificate is a {microshift-short} certificate authority (CA) then all of the signed certificates rotate. If you created any custom CAs, ensure the CAs manually rotate.
```

Certificate authority needs to be highlighted? https://redhat-documentation.github.io/supplementary-style-guide/#certificate-authority

. Short-lived certificates having certificate validity of one year.
. Long-lived certificates having certificate validity of 10 years.

Most server or leaf certificates are short-lived.

An example of a long-lived certificate is the client certificate for `system:admin user` authentication, or the certificate of the signer of the `kube-apiserver` external serving certificate.

[id="microshift-certificate-rotation_{context}"]
== Certificate rotation
Certificates that are expired or close to their expiration dates need to be rotated to ensure continued {microshift-short} operation. When {microshift-short} restarts for any reason, certificates that are close to expiring are rotated. A certificate that is set to expire imminently, or has expired, can cause an automatic {microshift-short} restart to perform a rotation.

[IMPORTANT]
====
If the rotated certificate is a {microshift-short} Certificate Authority, all of the certificates it signed rotate. If you created any Custom Certificate Authorities, they must be rotated manually.
====

[role="_additional-resources"]
.Additional resources

> * For more information about custom certificate authorities, see xref:../microshift_configuring/microshift_auth_security/microshift-custom-ca.adoc#microshift-custom-ca[Configuring custom certificate authorities].

Reviewer 3: Yes, this will have to go in the "things to know" assembly instead.

.. When a long-term certificate is 8.5 to 9 years old, it is rotated when {microshift-short} starts or restarts.

. Automatic restart for rotation:

> .. When a long-term certificate is more than 9 years old, {microshift-short} can automatically restart to rotate and apply a new certificate.

Reviewer 2: ```suggestion
.. When a long-term certificate is more than 9 years old, {microshift-short} might automatically restart so that it can rotate and apply a new certificate.
```
